File: flutter_app/lib/models/payment.dart

class Payment {
  final String id;
  final double amount;
  final String status; // 'pending', 'success', 'failed'
  final String? razorpayOrderId;
  final DateTime date;
  final String purpose;

  Payment({
    required this.id,
    required this.amount,
    required this.status,
    this.razorpayOrderId,
    required this.date,
    required this.purpose,
  });
}

File: backend/models/Payment.js
const mongoose = require('mongoose');

const PaymentSchema = new mongoose.Schema({
  id: { type: String, required: true, unique: true },
  user_id: { type: String, required: true, index: true },
  amount: { type: Number, required: true },
  status: { type: String, required: true },
  razorpayOrderId: String,
  date: { type: Date, required: true },
  purpose: { type: String, required: true }
}, { timestamps: true });

module.exports = mongoose.model('Payment', PaymentSchema);
